bundles/EcommerceFrameworkBundle/IndexService/Worker/ElasticSearch/AbstractElasticSearch.php

```diff
@@ -871,6 +873,135 @@ public function startReindexMode()
         }
     }
 
+
+    /**
+     *
+     * Perform a synonym update on the currently selected ES index, if necessary.
+     *
+     * Attention: the current index will be closed and opened, so it won't be available for a tiny moment (typically some milliseconds).
+     *
+     * @param string $indexNameOverride if given, then that index will be used instead of the current index.
+     * @param bool $skipComparison if explicitly set to true, then the comparison whether the synonyms between the current index settings
+     *        and the local index settings vary, will be skipped, and the index settings will be updated regardless.
+     * @param bool $skipLocking if explictly set to true, then no global lock will be activated / released.
+     *
+     * @throws \Exception is thrown if the synonym transmission fails.
+     */
+    public function updateSynonyms(string $indexNameOverride = "", bool $skipComparison = false, bool $skipLocking = true) {
+        try {
+
+            if (!$skipLocking) {
+                $this->activateIndexLock(); //lock all other processes
+            }
+
+            $indexName = $indexNameOverride ? : $this->getIndexNameVersion();
+
+            $indexSettingsSynonymPartLocalConfig = $this->extractMinimalSynonymFiltersTreeFromTenantConfig();
+            if (empty($indexSettingsSynonymPartLocalConfig)) {
+                Logger::info('No index update required, as no synonym providers are configured. '.
+                    'If filters have been removed, then reindexing will help to get rid of old configurations.'
+                );
+                return;
+            }
+
+            $esClient = $this->getElasticSearchClient();
+
+            if (!$skipComparison) {
+                $indexSettingsCurrentEs = $esClient->indices()->getSettings(['index' => $indexName])[$indexName]['settings']['index'];
+                $indexSettingsSynonymPartEs = $this->extractMinimalSynonymFiltersTreeFromIndexSettings($indexSettingsCurrentEs);
+
+                if (json_encode($indexSettingsSynonymPartEs) == json_encode($indexSettingsSynonymPartLocalConfig)) {
```

Comment:
compare arrays directly and not their json_strings ... should be faster.

Reply:
see also https://github.com/pimcore/pimcore/pull/6374#discussion_r427750038

Reply:
done.